flycheck-haskell.el

```diff
@@ -251,13 +251,21 @@ buffer."
   "Process the a Cabal CONFIG."
   (let-alist config
     (setq-local flycheck-ghc-search-path
-                (append .build-directories .source-directories
+                (append (if (string-equal (car flycheck-haskell-runghc-command)
+                                          "stack")
```

Comment:
@mrkkrp Ah, sorry, there's no relation between these at all.  This project is just about dumping the Cabal configuration, and we only use `stack` here because that's the safest way to get a `runghc` for Stack users—as you know Stack doesn't require a global GHC installation, and I presume many stack users don't have one anymore.  For them, `stack runghc` is actually the only way to get _any_ `runghc` to dum the Cabal configuration.

It doesn't matter whether Stack actually builds/checks this project or not, we just want to get any GHC to run our Haskell code to dump the Cabal configuration.

Adding all paths together is the simplest thing to do.  I think we should just try it, and look for issues.  If it causes problems we can still add a more sophisticated logic to differentiate between Stack and Cabal builds.  But if adding everything together turns out to be a solid approximation, we should we not keep that simple approach?